### PR TITLE
sorted fields by localized name/possibly inappropriate use of zope.i18n.translate()

### PR DESCRIPTION
**Pull request by zopyx**

### CODE REVIEW

File: plone/schemaeditor/fields.py [hunk 1]
```diff
@@ -61,7 +62,7 @@ def FieldsVocabularyFactory(context):
         terms.append(SimpleVocabulary.createTerm(factory,
                                                  translate(factory.title),
                                                  factory.title))
-
+    terms = sorted(terms, key=operator.attrgetter('token'))$
```
Comment (thet) [inline plone/schemaeditor/fields.py]: the dollar sign at the end of this line slipped in accidentally, not?

**Comment by davisagli** [comment]
This works in English, but not when translated.

screen shot 2015-04-12 at 9 33 04 am

**Comment by zopyx** [comment]
The problem is apparently that the translate() call

https://github.com/plone/plone.schemaeditor/blob/master/plone/schemaeditor/fields.py#L62

is completely useless and not working as expected. The translate() call
does not translate to the current language but just translates from EN TO EN
and the real translations seem to happen within the rendering pagetemplate.
That's why it is sorted by EN titles and appears unsorted with a different
language.

I encountered a similar problem with calling zope.i18n.translate() in my
own code and the only way around was to explicitly
parse the i18n domain and REQUEST.LANGUAGE into the translate() call.

Is there some related wrapped call to translate() within the Plone core or
is zope.i18n.translate() supposed to take
the 'plone' domain and REQUEST.LANGUAGE into account? I think not!?

2015-04-12 9:35 GMT-04:00 David Glick <notifications@github.com>:

**Comment by vincentfretin** [comment]
You always need to give the context like this:
translate(msg, context=self.request)

**Comment by vincentfretin** [comment]
If msg is not already a Message object, I usually do something like this:
PMF = MessageFactory('plone')
translate(PMF(msgstring), context=self.request)

**Comment by davisagli** [comment]
I updated this with @vincentfretin's recommendation and it seems to work. Merged it manually.

**Pull request closed without merging**